src/Symfony/Component/HttpKernel/DataCollector/DumpDataCollector.php

```diff
@@ -71,14 +72,17 @@ public function dump(Data $data)
         if ($this->stopwatch) {
             $this->stopwatch->start('dump');
         }
-        if ($this->isCollected && !$this->dumper) {
-            $this->isCollected = false;
-        }
 
         list('name' => $name, 'file' => $file, 'line' => $line, 'file_excerpt' => $fileExcerpt) = $this->sourceContextProvider->getContext();
 
-        if ($this->dumper) {
+        if ($this->dumper instanceof Connection) {
+            if (!$this->dumper->write($data)) {
```

Comment:
this is the root of the issue: not being able to know if we wrote anything to the server to set `$this->isCollected` correctly